### PR TITLE
chore(test): Upgrades Jest

### PR DESCRIPTION
**Pull request by johnpmitsch**
Upgrades jest package. Note that jest-cli provides jest. This is needed by react-testing-library https://github.com/testing-library/dom-testing-library/releases/tag/v7.0.0

### CODE REVIEW

**Comment by johnpmitsch** [comment]
@sharvit Here is the upgrade of jest. I didn't see any breaking changes that would affect us in the release notes, but could be missing something. I'm pretty sure the linking isn't using the right jest package in the CI.

Let me know if you have the same experience. I even tried running the jest command manually even from the `node_modules/@theforeman/...` path and it still seemed to use the older version.

If we can't figure out a way to test locally, we could discuss other options like merging and quickly following up with tests of the relevant repos like foreman and foreman-tasks.

File: packages/test/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "enzyme-adapter-react-16": "^1.15.2",
     "enzyme-to-json": "^3.4.3",
     "identity-obj-proxy": "^3.0.0",
-    "jest-cli": "^24.9.0",
+    "jest-cli": "^26.4.2",
```
Comment (sharvit) [inline packages/test/package.json:41]: Did you figure out why we are using jest-cli instead of jest?

Comment (johnpmitsch) [reply packages/test/package.json:41]: `jest` seems to be a wrapper of `jest-cli` [see here](https://www.npmjs.com/package/jest), and `jest-cli` provides all the same packages (`jest/core` and `import-local`) that `jest` does, plus it's own.

I don't see any official documentation about the difference, maybe it exists, but I assume they are interchangeable.

**Comment by johnpmitsch** [comment]
I opened up PRs for [foreman](https://github.com/theforeman/foreman/pull/7971) and [foreman-tasks](https://github.com/theforeman/foreman-tasks/pull/585) that fixes the breaking changes 

I was able to test and create new snapshots using [this method](https://github.com/theforeman/foreman-js/pull/225#issuecomment-688891393)

**Comment by johnpmitsch** [comment]
Updated to use just `jest` not `jest-cli` as discussed in the js meeting. They provide the same thing as discussed here https://github.com/theforeman/foreman-js/pull/212#discussion_r481133056 

See the dependencies here:
https://www.npmjs.com/package/jest
https://www.npmjs.com/package/jest-cli

**Comment by johnpmitsch** [comment]
@sharvit now that the tarball PR is merged, we can test this [using the GH action artifact](https://github.com/theforeman/foreman-js/blob/master/CONTRIBUTING.md#testing-pull-requests)

I also have https://github.com/theforeman/foreman/pull/7971 and https://github.com/theforeman/foreman-tasks/pull/585 to update the tests for jest 26

**Comment by johnpmitsch** [comment]
@amirfefer @sharvit any thoughts on getting this in? The accompanying foreman/foreman-tasks PRs should all be passing now

**Review by sharvit — APPROVED** [review]
Thanks @johnpmitsch, looks good and work as expected when using with the gh artifacts 👍

Because those changes break foreman and some plugins, I would like to release a new major version.

Version 5.0.0 already released and deprecated before so I am about to release 5.1 to npm manually.

**Comment by johnpmitsch** [comment]
thanks @sharvit, please let me know when the release is available and I can update the version in the foreman/foreman-tasks PRs and re-kick the tests!

**Comment by sharvit** [comment]
Just deployed 6.0.0, see: https://www.npmjs.com/package/@theforeman/test

**Pull request merged**